container.te

```diff
@@ -1033,3 +1033,33 @@ dontaudit container_domain device_node:chr_file setattr;
 dontaudit container_domain sysctl_type:file write;
 
 allow container_t proc_t:filesystem remount;
+
+
+########################################
+#
+# virt_launcher_t local policy
+#
+# SELinux security policy for kubevirt project
+# Temporary solution for rhbz#1795975
+
+type virt_launcher_t, container_domain;
+domain_type(virt_launcher_t)
+
+allow virt_launcher_t self:tun_socket { attach_queue relabelfrom relabelto create_socket_perms };
+
+manage_dirs_pattern(virt_launcher_t, container_var_run_t, container_var_run_t)
+manage_files_pattern(virt_launcher_t, container_var_run_t, container_var_run_t)
+
+manage_dirs_pattern(virt_launcher_t, container_share_t, container_share_t)
```

Comment:
Why does it need this access?

Reply:
I do not think this line is useful. I removed it from the module in https://github.com/kubevirt/kubevirt/pull/2934 -- which is still pending merge, but the fate of that PR is tied to this one.

I inspected the filesystems on random virt-launcher and virt-handler containers. The container_share_t label was not present at all. If it were present (and we really needed write access to it), that's much more likely to be properly fixed with relabelling than adding an allow-write rule to a type that's supposed to be read-only.

Reply:
As this is temporary solution, I allowed what was in previous cil policy. @fabiand mentioned that they will work on KubeVirt project to drop some access to make these containers more secure. No problem I can remove this line.

Reply:
We can create a type for virt-launcher to create in the policy than it would create that type when it runs in the /var/run directory.

type virt_launcher_var_run_t;
files_pid_file(virt_launcher_var_run_t)
files_pid_filetrans(virt_launcher_t, virt_launcher_var_run_t, { dir file lnk_file sock_file })
files_trans(virt_launcher_t, container_var_run_t, virt_launcher_var_run_t, dir)
manage_files_pattern(virt_launcher_t, virt_launcher_var_run_t, virt_launcher_var_run_t)
manage_lnk_files_pattern(virt_launcher_t, virt_launcher_var_run_t, virt_launcher_var_run_t)
manage_dirs_pattern(virt_launcher_t, virt_launcher_var_run_t, virt_launcher_var_run_t)

Reply:
Add a file context
/var/run/kubevirt-private(/.*)? 	gen_context(system_u:object_r:virt_launcher_var_run_t,s0)

Reply:
We could do similar for the content created in /tmp.

Reply:
Changed in the commit.